hack/ipa_debug_tools.sh

```diff
@@ -0,0 +1,227 @@
+        exit 1
+    fi
+
+    # Unzip ironic-python-agent.initramfs
+    sudo mkdir -p /tmp/debug-initramfs/build-ipa-initramfs
+    sudo cp /tmp/dib/ironic-python-agent.initramfs /tmp/debug-initramfs/
+    cd /tmp/debug-initramfs/build-ipa-initramfs || exit
+    gunzip -c ../ironic-python-agent.initramfs | sudo cpio -id
+
+    # Get master ironic-python-agent.initramfs and unzip
+    sudo mkdir -p /tmp/debug-initramfs/master-ipa-initramfs
+    cd /tmp/debug-initramfs || exit
+    sudo wget https://tarballs.opendev.org/openstack/ironic-python-agent/dib/ipa-centos9-master.tar.gz
+    sudo tar -xzf ipa-centos9-master.tar.gz
+    cd master-ipa-initramfs || exit
+    gunzip -c ../ipa-centos9-master.initramfs | sudo cpio -id
+}
+
+# -----------------------------------------------------------------------------
+# compare_dir_sizes <relative_path> [base_dir1] [base_dir2]
+#
+# Compares the sizes of files and directories at the given <relative_path>
+# inside two extracted initramfs directories (or any two directories).
+#
+# Usage examples:
+#   ./ipa_debug_tools.sh compare_dir_sizes
+#   ./ipa_debug_tools.sh compare_dir_sizes usr
+#   ./ipa_debug_tools.sh compare_dir_sizes usr/lib ~/ipa-build-initramfs ~/ipa-initramfs
+#   ./ipa_debug_tools.sh compare_dir_sizes . ~/ipa-build-initramfs ~/ipa-initramfs
+#
+# - <relative_path>: Path inside the compared directories (e.g., usr, usr/lib)
+# - [base_dir1]:     First base directory (default: /tmp/debug-initramfs/build-ipa-initramfs)
+# - [base_dir2]:     Second base directory (default: /tmp/debug-initramfs/master-ipa-initramfs)
+#
+# The function prints a table of entries, their sizes in both directories,
+# and the size difference. It also lists files only present in one directory.
+# -----------------------------------------------------------------------------
+compare_dir_sizes() {
+    path="${1:-}"
+    basedir1="${2:-/tmp/debug-initramfs/build-ipa-initramfs}"
+    basedir2="${3:-/tmp/debug-initramfs/master-ipa-initramfs}"
+
+    dir1="${basedir1}/${path}"
+    dir2="${basedir2}/${path}"
+    
+    # Color codes
+    RED='\033[0;31m'
+    GREEN='\033[0;32m'
+    YELLOW='\033[1;33m'
+    BLUE='\033[0;34m'
+    PURPLE='\033[0;35m'
+    CYAN='\033[0;36m'
+    NC='\033[0m' # No Color
+    BOLD='\033[1m'
+    
+    echo -e "${CYAN}📊 Comparing directories:${NC}"
+    echo -e "  ${BLUE}🏗️  Build IPA:${NC} ${dir1}"
+    echo -e "  ${GREEN}🎯 Master IPA:${NC} ${dir2}"
+    echo ""
+    
+    # Print table header
+    printf "${BOLD}%-40s %-22s %-22s %-22s${NC}\n" "Directory/File" "🏗️  Build" "🎯 Master" "📏 Diff"
+    echo "======================================================================================================="
+
+    tmpfile=$(mktemp)
+    checked=0
+    skipped=0
+
+    # Compare entries present in both directories
+    find "${dir1}" -mindepth 1 -maxdepth 1 -printf '%f\n' | sort > /tmp/dir1_list
+    find "${dir2}" -mindepth 1 -maxdepth 1 -printf '%f\n' | sort > /tmp/dir2_list
```

Comment:
Suggestion: Instead of hardcoded temp file paths, maybe use `mktemp`  to create random tmp files. This allows running the script in two terminals without conflict, which I think might be useful at times.